### ISSUE TITLE
[Help] Global Multivariate Forecasting

### ISSUE THREAD

**Issue by juste97** [unit8co/darts#1984]
Hey there,

I'm having trouble with the global models, especially how to prepare the data and what to use for predictions. Maybe someone can point me in the right direction. I'm using the Walmart dataset from Kaggle right now.

What I've done so far is split the data before grouping with **from_group_dataframe** and then make separate time series grouped by "store" for each target and the convariates for both training and testing sets and scale them.

Problems seem to start with the scaler; when plotting both training and testing, in the end they both seem to be scaled differently. Am I supposed to fit the scaler on the whole dataset before splitting?

Also, regarding **past_covariates** in **predict** should this include the data for both the training and testing sets? When using the test set for the covariates (x_test), I've received an error that the data does not go back far enough.

For the series used in **predict**, I have already figured out that this needs to be the training set.

Big thanks for this package; the available models and data prep options look really nice, and I can't wait to try out more different things once I get this to work 👍 

```python

df_train = df.loc[df["Date"] < train_test_split_date]
df_test = df.loc[df["Date"] >= train_test_split_date]

y_train = TimeSeries.from_group_dataframe(df_train, group_cols="Store", time_col="Date", value_cols=["Weekly_Sales"],
                                         freq="W")
x_train = TimeSeries.from_group_dataframe(df_train, group_cols="Store", time_col="Date", value_cols=features, freq="W")
y_test = TimeSeries.from_group_dataframe(df_test , group_cols="Store",time_col="Date", value_cols=["Weekly_Sales"],
                                         freq="W")
x_test = TimeSeries.from_group_dataframe(df_test , group_cols="Store", value_cols=features, freq="W")

y_scaler_dataset = Scaler()
x_scaler_dataset = Scaler()

y_train = y_scaler_dataset.fit_transform(y_train)
x_train = x_scaler_dataset.fit_transform(x_train)

y_test = y_scaler_dataset.fit_transform(y_test)
x_test = x_scaler_dataset.fit_transform(y_test)

forecast_steps=10
epochs=200

model_BlockRNNModel = BlockRNNModel(
     model="RNN",
     input_chunk_length=15,
     output_chunk_length=forecast_steps,
     n_epochs=epochs,
     random_state=0,
 )

model_BlockRNNModel.fit(
    series=y_train,
    past_covariates=x_train,
    verbose=True,
)

pred_cov = model_BlockRNNModel.predict(n=forecast_steps, series=y_train, past_covariates=x_train )
}

y_test = y_scaler_dataset.inverse_transform(y_test)
y_train = y_scaler_dataset.inverse_transform(y_train )
pred_cov = y_scaler_dataset.inverse_transform(pred_cov)

```

**Comment by madtoinou** [comment]
Hi @juste97,

If you pass a sequence of `TimeSeries` to any Darts' fittable data tranformer, it will fit each series independently. In order to fit and transform all your shops' series with a single scaler, you need to apply it before splitting the dataset. However, scaling them independently could be preferable depending on your dataset, especially if they have very different ranges (otherwise, shops with small volumes will be scaled to values very close to 0 because of the large stores).

You can use the same `past/future_covariates` series for both training and validation, Darts will take care of slicing away the timestamps it does not need. If you want to split them, you'll need to account for the lags of the model and my need to shift the start/end of each covariates series.

The series passed to `predict()` is usually the training series because models usually look at past values, so to predict the first value of the validation series, you actually need the last timestamps of the training series. It also helps the model "when" is the forecasting actually starting.

**Comment by dennisbader** [comment]
To add to @madtoinou's reponse: after fitting the data transofrmers on the train set, you only need to transform the values of the test set using the fitted tranformers. 
You need to use:

```
y_test = y_scaler_dataset.transform(y_test)
x_test = x_scaler_dataset.transform(x_test)
```

You can then concatenate x_train and x_test to `x` and use `past_covariates=x` in both fit and predict.

```
from darts import concatenate
x = [concatenate([x_tr, x_te]) for x_tr, x_te in zip(x_train, x_test)]
```

**Comment by juste97** [comment]
Sorry for the late reply. This helped a lot. Thank you both!

@madtoinou I've also found global_fit, which can be passed in the Scaler class. This seems to pretty much do what you described.

However, regarding scaling before splitting the data, doesn't this introduce leakage (by using the mean and standard deviation of the testing set as well)?

**Comment by madtoinou** [comment]
`global_fit` is indeed the best way to fit a single scaler on a sequence of series.

I meant before splitting into several series, indeed the fitting should be performed only of the training set and then applied to the validation/test in order to avoid data leakage.
